src/libraries/Common/src/System/Net/SocketAddressPal.Windows.cs

```diff
@@ -26,80 +27,44 @@ public static unsafe void SetAddressFamily(byte[] buffer, AddressFamily family)
             }
 
 #if BIGENDIAN
-            buffer[0] = unchecked((byte)((int)family >> 8));
-            buffer[1] = unchecked((byte)((int)family));
+            BinaryPrimitives.WriteUInt16BigEndian(buffer, (ushort)family);
 #else
-            buffer[0] = unchecked((byte)((int)family));
-            buffer[1] = unchecked((byte)((int)family >> 8));
+            BinaryPrimitives.WriteUInt16LittleEndian(buffer, (ushort)family);
 #endif
         }
 
         public static unsafe ushort GetPort(ReadOnlySpan<byte> buffer)
-        {
-            return buffer.NetworkBytesToHostUInt16(2);
-        }
+            => BinaryPrimitives.ReadUInt16BigEndian(buffer.Slice(2));
 
         public static unsafe void SetPort(byte[] buffer, ushort port)
-        {
-            port.HostToNetworkBytes(buffer, 2);
-        }
+            => BinaryPrimitives.WriteUInt16BigEndian(buffer.AsSpan(2), port);
 
         public static unsafe uint GetIPv4Address(ReadOnlySpan<byte> buffer)
-        {
-            unchecked
-            {
-                return (uint)((buffer[4] & 0x000000FF) |
-                    (buffer[5] << 8 & 0x0000FF00) |
-                    (buffer[6] << 16 & 0x00FF0000) |
-                    (buffer[7] << 24));
-            }
-        }
+            => BinaryPrimitives.ReadUInt32LittleEndian(buffer.Slice(4));
 
         public static unsafe void GetIPv6Address(ReadOnlySpan<byte> buffer, Span<byte> address, out uint scope)
         {
-            for (int i = 0; i < address.Length; i++)
-            {
-                address[i] = buffer[8 + i];
-            }
+            buffer.Slice(8, address.Length).CopyTo(address);
 
-            scope = unchecked((uint)(
-                (buffer[27] << 24) +
-                (buffer[26] << 16) +
-                (buffer[25] << 8) +
-                (buffer[24])));
+            scope = BinaryPrimitives.ReadUInt32LittleEndian(buffer.Slice(24));
         }
 
         public static unsafe void SetIPv4Address(byte[] buffer, uint address)
         {
             // IPv4 Address serialization
-            buffer[4] = unchecked((byte)(address));
-            buffer[5] = unchecked((byte)(address >> 8));
-            buffer[6] = unchecked((byte)(address >> 16));
-            buffer[7] = unchecked((byte)(address >> 24));
+            BinaryPrimitives.WriteUInt32LittleEndian(buffer.AsSpan(4), address);
         }
 
         public static unsafe void SetIPv6Address(byte[] buffer, Span<byte> address, uint scope)
         {
             // No handling for Flow Information
-            buffer[4] = (byte)0;
-            buffer[5] = (byte)0;
-            buffer[6] = (byte)0;
-            buffer[7] = (byte)0;
+            BinaryPrimitives.WriteUInt32LittleEndian(buffer.AsSpan(4), 0);
```

Comment:
Endianness here wouldn't matter.  But I guess it doesn't really matter perf-wise, especially on a little-endian arch.